src/crawler/crawl.ts

```diff
@@ -233,7 +234,9 @@ class Crawler {
       }
 
       this.publicKeysSeen.add(normalizedPublicKey)
-      promises.push(saveNode(dbNode))
+
+      // If a peer has the node IP as null, we want to overwrite in the database
+      promises.push(saveNode(dbNode, true))
```

Comment:
@achowdhry-ripple Can you please check if only this change will fix the issue as suggested by @pdp2121 as well. 
We can remove boolean from the saveNode method and saveNode remains the same as its in main branch?

```suggestion
      promises.push(saveNode({ ip: null, port: null, ...dbNode}))
```

Reply:
@Patel-Raj This would ruin the typing for the Node object, and require me to change `ip` and `port` to allow for null which breaks other parts of the code for me that reference these parameters. Personally I found this more messy and would rather update `saveNode` than make the typing here more opaque. Can discuss further if you feel strongly about this, or if I missed something and you were able to make this change a different way that doesn't require typing changes?

Reply:
@achowdhry-ripple Ya, I had to change the type for ip and port and add a null check just below this line.

`promises.push(saveNode({ ip: null, port: null, ...dbNode }))`

```
ip?: string | null
port?: number | null
```

You are right, this also, gets a little messy but keeps the saveNode clean as its purpose is to save or update the node if it exists. No, strong opinions.

Reply:
true, I'd rather keep it like this for typing clarity in general, since this essentially adds a toggle for modifying the merge functionality in the db save which could be extensible in the future if there are more asks like this (other fields that should be nullable). If anyone strongly disagrees and would rather have it the other way, am open to discuss but will otherwise close out this thread and merge soon